src/content-helper/dashboard-widget/provider.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ContentHelperError,
+	ContentHelperErrorCode,
+} from '../../blocks/content-helper/content-helper-error';
+import { TopPostData } from './top-posts/model';
+import {
+	convertDateToString,
+	removeDaysFromDate,
+} from '../../blocks/shared/utils/date';
+
+/**
+ * The form of the response returned by the /stats/posts WordPress REST API
+ * endpoint.
+ */
+interface TopPostsApiResponse {
```

Comment:
Instead of defining new interface I think it is better to extend the existing interface i.e. if we are using same API then interfaces should also be common.

Reply:
👍 I'm planning a refactoring PR after this one, which should harmonize Content Helper code a bit more. We have made some progress for re-use and will be continuing that.

Reply:
I think it is better to do in same PR .... but your call 🙂